core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -96,45 +114,76 @@ oppia.controller('FeedbackTab', [
       backdrop: true,
       size: 'lg',
       resolve: {
-        suggestion: function() {
-          return $scope.activeThread.suggestion;
+        suggestionInfo: function() {
+          return {
+            isSuggestionValid: $scope.isSuggestionValid(),
+            hasUnsavedChanges: $scope.hasUnsavedChanges()
+          };
+        },
+        stateContent: function() {
```

Comment:
(In contrast to the previous comment, the key-value pairs in this one are more conceptually related, so I think it's OK as-is, although separate entries would be fine as well.)